### PR TITLE
DDEV-install: fixes table prefix when using `-d` and `-s` flag

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

When using ddev command `openmage-install` with `-d` flag (use default values) combined with `-s` (sample data) table prefix is set wrong and you'll see no sample data.

### Manual testing scenarios (*)
1.  `ddev openmage-install -s -d -k`

### CODE REVIEW

**Comment by sreichel** [comment]
@ADDISON74 requst review knowing you use ddev.

**Comment by ADDISON74** [comment]
I did a checkout on this PR. Here is the Terminal window content after running the command

```ddev openmage-install -s -d -k```

```php
ubuntu@DESKTOP-GBHMUPN:~/openmage$ ddev openmage-install -s -d -k
OpenMage is already installed. Delete local.xml and drop the database? [y/N] y
Uncompressing Sample Data...
Copying Sample Data into the OpenMage directory...
Importing Sample Data into the database...
PHP Fatal error:  Uncaught Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:510
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(318): Mage_Core_Model_Config->getCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}

Next Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:534
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(326): Mage_Core_Model_Config->releaseCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}
  thrown in /var/www/html/app/code/core/Mage/Core/Model/Config.php on line 534

Fatal error: Uncaught Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:510
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(318): Mage_Core_Model_Config->getCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}

Next Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:534
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(326): Mage_Core_Model_Config->releaseCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}
  thrown in /var/www/html/app/code/core/Mage/Core/Model/Config.php on line 534
Failed to run openmage-install -s -d -k: exit status 255
```

If I am reloading the Frontend page in the browser I am getting this content

```php
Fatal error: Uncaught Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:510 Stack trace: #0 /var/www/html/app/code/core/Mage/Core/Model/App.php(436): Mage_Core_Model_Config->getCacheSaveLock() #1 /var/www/html/app/code/core/Mage/Core/Model/App.php(347): Mage_Core_Model_App->_initModules() #2 /var/www/html/app/Mage.php(743): Mage_Core_Model_App->run(Array) #3 /var/www/html/index.php(56): Mage::run('', 'store') #4 {main} Next Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:534 Stack trace: #0 /var/www/html/app/code/core/Mage/Core/Model/App.php(448): Mage_Core_Model_Config->releaseCacheSaveLock() #1 /var/www/html/app/code/core/Mage/Core/Model/App.php(347): Mage_Core_Model_App->_initModules() #2 /var/www/html/app/Mage.php(743): Mage_Core_Model_App->run(Array) #3 /var/www/html/index.php(56): Mage::run('', 'store') #4 {main} thrown in /var/www/html/app/code/core/Mage/Core/Model/Config.php on line 534
```

I checked out the main branch and this is the content of the Terminal window

```php
ubuntu@DESKTOP-GBHMUPN:~/openmage$ ddev openmage-install -s -d -k
Uncompressing Sample Data...
Copying Sample Data into the OpenMage directory...
Importing Sample Data into the database...
PHP Fatal error:  Uncaught Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:510
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(318): Mage_Core_Model_Config->getCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}

Next Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:534
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(326): Mage_Core_Model_Config->releaseCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}
  thrown in /var/www/html/app/code/core/Mage/Core/Model/Config.php on line 534

Fatal error: Uncaught Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:510
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(318): Mage_Core_Model_Config->getCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}

Next Error: Call to a member function fetchOne() on bool in /var/www/html/app/code/core/Mage/Core/Model/Config.php:534
Stack trace:
#0 /var/www/html/app/code/core/Mage/Core/Model/Config.php(326): Mage_Core_Model_Config->releaseCacheSaveLock()
#1 /var/www/html/app/code/core/Mage/Core/Model/App.php(270): Mage_Core_Model_Config->init(Array)
#2 /var/www/html/app/Mage.php(676): Mage_Core_Model_App->init('default', 'store', Array)
#3 /var/www/html/install.php(114): Mage::app('default')
#4 {main}
  thrown in /var/www/html/app/code/core/Mage/Core/Model/Config.php on line 534
Failed to run openmage-install -s -d -k: exit status 255
```
I never used flags just ```ddev openmage-install``` I didn't encounter any problems. Right now I am trying to find a procedure to get back to the initial stage before running the command with flags.

**Comment by sreichel** [comment]
Thats not related to the script itself ... you have to clear `var/cache` directory.

Maybe i should add this too.

**Comment by ADDISON74** [comment]
I was preparing to write you the same thing. This situation must be foreseen by deleting {cache|session|report|lock} directories from /var/log. If we don't do it we will get that error.

**Comment by ADDISON74** [comment]
If {cache | session} directories exist then they must be deleted first. Please add this action to the script.

**Comment by sreichel** [comment]
Before i fix it ....

1. set maintenance.flag
2. clear cache
3. run install script
4. delete maintenance.flag

???

**Comment by ADDISON74** [comment]
It is much better using the maintenance flag.

**Comment by ADDISON74** [comment]
I tested the initial DDEV command, not the version modified by this PR.

- I dropped the database
- I created a new one called db
- I deleted /var/log/{cache|session} directories
- I successfully ran the command in Terminal ```ddev openmage-install -s -d -k``` (sample / default / keep)

In the PR description you say that using the command with the flags -d and -s, the table prefix is set incorrectly and the products from the sample package are not displayed. In the database all tables have no prefix. I loaded the Frontend in the browser and all the sample products are displayed. 

At first glance, the initial DDEV command would not have any problem. Am I wrong in the assessment? However, if we look at the script, the proposed change is logical as it should be.

**Comment by sreichel** [comment]
> I loaded the Frontend in the browser and all the sample products are displayed.

Strange. I can reproduce it with every install try.

Updated PR works flawless for me. Only with dev-mode ON i run into https://github.com/OpenMage/magento-lts/pull/3677.

File: .ddev/commands/web/openmage-install [hunk 1]
```diff
@@ -87,11 +90,15 @@ if [[ $INSTALL_SAMPLE_DATA =~ ^(yes|y) ]]; then
   echo "Copying Sample Data into the OpenMage directory..."
   cp -r magento-sample-data-1.9.2.4/* "${ROOT}/"
 
+  echo "Clear var/cache ..."
```
Comment (ADDISON74) [inline .ddev/commands/web/openmage-install:93]: Clearing /var/cache directory... 

Just for keeping the same format and and punctuation.

File: .ddev/commands/web/openmage-install [hunk 2]
```diff
@@ -40,6 +40,9 @@ while :; do
      shift
  done
 
+echo "Enable maintenance mode ..."
```
Comment (ADDISON74) [inline .ddev/commands/web/openmage-install:43]: Enable maintenance mode ...

Delete the space between mode and ...

File: .ddev/commands/web/openmage-install [hunk 3]
```diff
@@ -152,3 +163,6 @@ php -f install.php -- \
   --default_currency 'USD' \
   --enable_charts 'yes' \
   --skip_url_validation 'yes'
+
+echo "Disable maintenance mode ..."
```
Comment (ADDISON74) [inline .ddev/commands/web/openmage-install:167]: Remove the blank space between mode and ...

**Review by ADDISON74 — CHANGES_REQUESTED** [review]

**Comment by ADDISON74** [comment]
I deleted everything from the file system and from Docker. I cloned the OpenMage repository, I ran the following commands in order:

```ddev config```
```ddev composer install```
```ddev start```

If I load the Frontend it is fine, there are no categories and products.

Now I run the following command in Terminal:

```php
ubuntu@DESKTOP-GBHMUPN:~/openmage$ ddev openmage-install -s -d -k
OpenMage is already installed. Delete local.xml and drop the database? [y/N] y
Creating Sample Data directory...
Downloading Sample Data...
--2024-02-01 19:48:45--  https://github.com/Vinai/compressed-magento-sample-data/raw/master/compressed-magento-sample-data-1.9.2.4.tgz
Resolving github.com (github.com)... 140.82.121.3
Connecting to github.com (github.com)|140.82.121.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/Vinai/compressed-magento-sample-data/master/compressed-magento-sample-data-1.9.2.4.tgz [following]
--2024-02-01 19:48:45--  https://raw.githubusercontent.com/Vinai/compressed-magento-sample-data/master/compressed-magento-sample-data-1.9.2.4.tgz
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.109.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 68614817 (65M) [application/octet-stream]
Saving to: ‘sample_data.tgz’

sample_data.tgz            100%[========================================>]  65.44M   756KB/s    in 82s

2024-02-01 19:50:13 (817 KB/s) - ‘sample_data.tgz’ saved [68614817/68614817]

Uncompressing Sample Data...
Copying Sample Data into the OpenMage directory...
Importing Sample Data into the database...
SUCCESS: 6e6407d09d97a4a4bd1e39497ebfd70a
```

If I load the Frontend, nothing has changed. I have no categories, I have no products. so I was able to reproduce the problem.

I checked out on this PR

```php
ubuntu@DESKTOP-GBHMUPN:~/openmage$ ddev openmage-install -s -d -k
Enable maintenance mode ...
OpenMage is already installed. Delete local.xml and drop the database? [y/N] y
Uncompressing Sample Data...
Copying Sample Data into the OpenMage directory...
Clear var/cache ...
Importing Sample Data into the database...
SUCCESS: 21307e631c8891b6fcae5b9194257cee
Disable maintenance mode ...
```

If I load the Frontend, I have the content on the page from the Magento Sample package. My opinion that PR solves the problem.

**Comment by ADDISON74** [comment]
@sreichel - Please make the requested changes in the sentences to approve this PR.

I did not identify this issue before because I always used the command ```ddev openmage-install``` without flags. I am glad it is fine now. DDEV helped me a lot to test OpenMage and I really recommend everyone to do it because it is extremely simple to get a test environment in which everything is quickly configurable, from webser, php version, database and many others.

**Review by ADDISON74 — APPROVED** [review]

**Comment by sreichel** [comment]
Note: to not get asked for deleting local.xml and drop DB, you can use `-q` ;)

(`-k` to keep downloaded saple data is still optional)

**Comment by ADDISON74** [comment]
We need a green approval for this PR. Whoever gives it doesn't have to worry because only two people will use this change.

**Comment by sreichel** [comment]
> because only two people will use this change

Does really nobody else use ddev?

**Comment by fballiano** [comment]
is this PR finished? in that case I'll merge it

**Comment by sreichel** [comment]
Can be merged

**Pull request merged**